ts/packages/anchor/tests/types.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { IdlTypes } from "../src/program/namespace/types";
+
+type IdlWithEnum = {
+  version: "0.1.0";
+  name: "blockchain";
+  instructions: [];
+  types: [
+    {
+      name: "Currency";
+      type: {
+        kind: "enum";
+        variants: [
+          {
+            name: "Sol";
+          },
+          {
+            name: "Usdc";
+          }
+        ];
+      };
+    }
+  ];
+};
+
+describe("types", () => {
+  describe("IdlTypes", () => {
+    it("covers enum with only name", () => {
+      type Currency = IdlTypes<IdlWithEnum>["Currency"];
+      typeCheck<
```

Comment:
🍎 if the first argument will not be equals second argument you will get a typescript